File: Remove Element/Remove_Element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int i =0,j = 0;
        int n = nums.size();
        while(j<n) {
            if(nums[j] != val) {
                nums[i] = nums[j];
                j++;
                i++;
            }
            else j++;
        }
        return i;
    }
};

File: Remove Element/Remove_Element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int first = 0,second = nums.length-1;
        while(first <= second) {
            if(nums[first] == val) {
                int temp = nums[first];
                nums[first] = nums[second];
                nums[second] = temp;
                second--;
            }
            else first++;
        }
        return first;
    }
}